app/coffeescripts/views/conferences/ConferenceView.coffee

```diff
@@ -126,3 +128,83 @@ define [
         else
           window.open(data[0].url)
       )
+
+    deleteRecording: (e) ->
+      e.preventDefault()
+      if !confirm I18n.t("Are you sure you want to delete this recording?")
+        $(e.currentTarget).parents('.inline-block').find('.al-trigger').focus()
+        return
+      $deleteButton = $(e.currentTarget).parent()
+      @toggleActionButton($deleteButton, {state: "processing", action: "delete"})
+      @toggleRecordingLink($deleteButton, {state: "processing"})
+      $.ajaxJSON($deleteButton.data('url') + "/recording", "DELETE", {
+          recording_id: $deleteButton.data("id"),
+        }, (data) =>
+          if data.deleted == "true"
+            @removeRecordingRow($deleteButton)
+            return
+          @ensureActionPerformed($deleteButton, 1, @removeRecordingRow)
+      )
```

Comment:
I mean this is okay I guess but I would rather have a .then and catch rather than just a data that returns that we check for true but I won't -1 on that.

Reply:
I see what you mean. The only point of failure for the client is if Canvas goes down. This is because the controller/model are taking care of giving a response true/false to this ajax request.

What we can do to make it more resilient is to add a fail method with the corresponding code for failing over. Would that be OK?

Reply:
@jfederico That would be great!